src/rnp/rnp.cpp

```diff
@@ -676,6 +682,12 @@ rnp_main(int argc, char **argv)
     }
 
     if (!cli_rnp_check_weak_hash(&rnp)) {
+        if (rnp::str_case_eq(rnp.cfg().get_hashalg(), "SHA1")) {
```

Comment:
No need to make check here as `--allow-weak-hash` applies to the encryption/signing, not to the key signatures.